modules/compliance-tests/src/smithy4s/compliancetests/ReverseRouter.scala

```diff
@@ -19,17 +19,18 @@ package smithy4s.compliancetests
 import smithy4s.Service
 import cats.effect.Resource
 import smithy4s.kinds.FunctorAlgebra
-import smithy4s.http.CodecAPI
+import smithy4s.http.HttpMediaType
 import smithy4s.ShapeTag
 import org.http4s.HttpApp
+import smithy4s.schema.Schema
 
 /* A construct encapsulating the action of turning an http4s route into
  * an an algebra
  */
 trait ReverseRouter[F[_]] {
   type Protocol
   def protocolTag: ShapeTag[Protocol]
-  def codecs: CodecAPI
+  def expectedResponseType(schema: Schema[_]): HttpMediaType
```

Comment:
any reason why the schema is required here?

Reply:
Ah I guess the `@mediaType` annotation?

Reply:
> Ah I guess the @mediaType annotation?

Indeed :) but also some protocols might have different content types based on other factors. Like the AWS Rest Json protocol interpret blob payloads as binary and string payloads as text.